packages/lyra/src/lyra.ts

```diff
@@ -97,8 +126,13 @@ export class Lyra<TSchema extends PropertiesSchema = PropertiesSchema> {
 
       if (isNested) {
         this.buildIndex(schema[prop] as TSchema, `${propName}.`);
-      } else {
+      } else if (propValue === "string") {
```

Comment:
Numeric comparisons are faster than string based comparisons. What about having a constant of ValueType.STRING with a value of number?

Reply:
I am not getting what you are suggesting here, can you please elaborate?